src/share/streams/shr_stream_mod.F90

```diff
@@ -257,7 +257,11 @@ subroutine shr_stream_init(strm,infoFile,yearFirst,yearLast,yearAlign,taxMode,rc
     !--- read data ---
     read(nUnit,'(a)',END=999) str
     call shr_string_leftalign_and_convert_tabs(str)
-    strm%dataSource = str
+    if(len_trim(str)<= len(strm%dataSource)) then
+       strm%dataSource = str(1:len(strm%dataSource))
+    else
+       call shr_sys_abort('dataSource too long for variable ' // errMsg(sourcefile, __LINE__))
```

Comment:
Sorry, my comment wasn't clear here. `Using errMsg(sourcefile, __LINE__)` will only work if you define sourcefile and add a use statement for errMsg. Why don't you just change this to match the others:

```Fortran
call shr_stream_abort(subName//" ERROR: dataSource too long for variable")
```